### PR TITLE
Bump apalache version

### PR DESCRIPTION
**Pull request by bugarela**
Hello :octocat: 

This is a bump for apalache, as a new version was released today and it is relevant to Quint.

Among other fixes, the compiled TLA+ modules are now type annotated, so I also updated the integration test expectations.

### CODE REVIEW

File: quint/io-cli-tests.md [hunk 1]
```diff
@@ -24,11 +24,15 @@ We want to ensure we do not throw uncaught exceptions when the input file is
 doesn't exist.
 
 <!-- !test in non-existent file -->
-    quint parse ../examples/non-existent.file
+```
```
Comment (bugarela) [inline quint/io-cli-tests.md:27]: unrelated stylistic changes

**Review by p-offtermatt — APPROVED** [review]
LGTM
